lib/metrics/StatsClient.js

```diff
@@ -65,15 +65,20 @@ class StatsClient {
     /**
     * report/record a new request received on the server
     * @param {string} id - service identifier
+    * @param {number} amount - amount to increment by (default = 1)
     * @param {callback} cb - callback
     * @return {undefined}
     */
-    reportNewRequest(id, cb) {
+    reportNewRequest(id, amount = 1, cb) {
```

Comment:
You should make sure the existing API is not broken with the changes, here the existing callers that may provide `cb` still expect their callback to be called.

I saw this pattern already in some standard node code, where arguments with default values are not the trailing arguments, I think it's fine but requires proper handling in the function. I would think a proper way to deal with this in a clean way would be to declare the arguments with spread syntax `...args`, pop the last argument as the callback function if it's of type `function`, otherwise skip it (consider callback is not provided), and keep going with the preceding args to fit the exposed API. Then any existing caller of `reportNewRequest(id, cb)` would not be broken.

Another more straightforward option is to create a new API function with an additional mandatory `amount` argument.

Reply:
Sorry I missed your comment in the PR about this :)

You can still implement the above or leave as is, up to you, but if you don't you should fix the test(s) that use the old API with a callback, see my comment below :)